BinarySearch/MedianOfArray.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+There are two sorted arrays A and B of size m and n respectively.
+
+Find the median of the two sorted arrays ( The median of the array formed by merging both the arrays ).
+
+The overall run time complexity should be O(log (m+n)).
+
+Sample Input
+
+A : [1 4 5]
+B : [2 3]
+
+Sample Output
+
+3
+
+ NOTE: IF the number of elements in the merged array is even, then the median is the average of n / 2 th and n/2 + 1th element.
+For example, if the array is [1 2 3 4], the median is (2 + 3) / 2.0 = 2.5
+
+https://www.interviewbit.com/problems/median-of-array/
+
+*/
+
+double Solution::findMedianSortedArrays(const vector<int> &A, const vector<int> &B) {
+    // Do not write main() function.
+    // Do not read input, instead use the arguments to the function.
+    // Do not print the output, instead return values as specified
+    // Still have a doubt. Checkout www.interviewbit.com/pages/sample_codes/ for more details
+    if(A.empty())
+        return (B.size()%2==0)? (B[B.size()/2]+B[B.size()/2-1])/2.0: B[B.size()/2];
```

Comment:
1. Lines `30` and `32` are duplicated code which can be moved to a separate helper function and supplied vectors A or B.
1. The code is not legible, could you please simplify the expression, why are we dividing with (2-1) which will always result in a constant ? Please try and simplify the whole expression and add code comment explaining what's being done in the expression.